src/main/java/org/interledger/ilqp/QuoteLiquidityResponse.java

```diff
@@ -48,5 +48,148 @@
    * @return An instance of {@link ZonedDateTime}.
    */
   ZonedDateTime getExpiresAt();
+  
+  /**
+   * A builder for instances of {@link QuoteLiquidityResponse}.
+   */
+  class Builder {
+    private LiquidityCurve liquidityCurve;
+    private InterledgerAddress appliesTo;
+    private Duration sourceHoldDuration;
+    private ZonedDateTime expiresAt;
+    
+    /**
+     * Set the liquidity curve into this builder.
+     *
+     * @param liquidityCurve To An instance of {@link LiquidityCurve}.
+     */
+    public Builder liquidityCurve(final LiquidityCurve liquidityCurve) {
+      this.liquidityCurve = Objects.requireNonNull(liquidityCurve);
```

Comment:
The way I wanted to differentiate between failing early and failing during `build()` was based on whether the supplied object was mutable after being provided.

Thinking about that, it obviously still has to be non-null so I agree 👍 

Need to comb through other builders and ensure we're being consistent.